src/interfacers/EmonHubBleInterfacer.py

```diff
@@ -0,0 +1,168 @@
+import time
+import struct
+from bluepy import btle
+import Cargo
+from emonhub_interfacer import EmonHubInterfacer
+
+"""class EmonhubBleInterfacer
+
+Polls a Bluetooth LE sensor for temperature, huimidity and battery level
+
+Currently only tested with a Silicon Labs Thunderboard Sense 2
+
+
+Example config snippets:
+
+[interfacers]
+
+[[blesensor]]
+    Type = EmonHubBleInterfacer
+    [[[init_settings]]]
+        device_addr = '00:0b:57:64:8c:a2'
+    [[[runtimesettings]]]
+        pubchannels = ToEmonCMS,
+	read_interval = 20
+
+[nodes]
+
+[[1]]
+    nodename = Sensornode
+
+    [[[rx]]]
+        names = temp,humidity,battery
+        scales = 0.01,0.01,1
+        units = C,%,%
+
+"""
+
+class EmonHubBleInterfacer(EmonHubInterfacer):
+
+    def __init__(self, name, device_addr=''):
+        """Initialize interfacer
+
+        device_addr (string): BLE MAC address to connect to
+
+        """
+
+        # Initialization
+        super(EmonHubBleInterfacer, self).__init__(name)
+
+        self._private_settings = {
+            'read_interval': 60
+        }
+
+        self._addr = device_addr
+        self._last_read_time = 0
+        self._bat_readings = []
+
+	self._connect()
+
+    def close(self):
+        """Close serial port"""
+
+        # Close serial port
+        if self._ble is not None:
+            self._log.debug("Closing Bluetooth connection")
+            self._ble.disconnect()
+
+        return
```

Comment:
This line doesn't do anything. Remove.

Reply:
Should it be:

        if self._ble:
            self._log.debug("Closing Bluetooth connection")
            self._ble.disconnect()